### PR TITLE
First commit of NLPModels.jl split into several smaller packages

### PR DESCRIPTION
**Pull request by abelsiqueira**
Pulling into a new branch `breakup` to check travis

### CODE REVIEW

File: src/NLSModels.jl [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export AbstractNLSModel, nls_meta, NLSCounters, reset!,
-       residual, residual!, jac_residual, jac_structure_residual, jac_coord_residual!,
+       residual, residual!, jac_residual, jac_structure_residual!, jac_structure_residual, jac_coord_residual!,
```
Comment (dpo) [inline src/NLSModels.jl]: Would it be too much asking to split this into a separate PR to keep things contained?

File: test/runtests.jl [hunk 2]
```diff
@@ -1,16 +1,18 @@
 using Test, NLPModels, LinearAlgebra, LinearOperators, Printf, SparseArrays
 
+const problems = ["BROWNDEN", "HS5", "HS6", "HS10", "HS11", "HS14"]
+
 # Including problems so that they won't be multiply loaded
-for problem in [:brownden, :genrose, :hs5, :hs6, :hs10, :hs11, :hs14]
+for problem in Symbol.(lowercase.(problems))
```
Comment (dpo) [inline test/runtests.jl]: Nice!

File: test/test_lls_model.jl [hunk 3]
```diff
@@ -24,6 +24,62 @@ function lls_test()
       @test nls.meta.nnln == length(nls.meta.nln) == 0
     end
   end
+
+  @testset "Consistency of Linear problem" begin
```
Comment (dpo) [inline test/test_lls_model.jl]: Does this also belong in a separate PR?

File: src/NLSModels.jl (hunk 1 above)
Comment (abelsiqueira) [reply src/NLSModels.jl]: Absolutely. I plan to separate this and some other small things, but I made the PR to test.

**Comment by abelsiqueira** [comment]
Closing as per #199
